esphome/components/http_request/__init__.py

```diff
@@ -78,6 +82,15 @@ def validate_secure_url(config):
             cv.Optional(
                 CONF_TIMEOUT, default="5s"
             ): cv.positive_time_period_milliseconds,
+            cv.Optional(CONF_RETRY, default={}): cv.Schema(
+                {
+                    cv.Optional(CONF_COUNT, 0): cv.int_,
+                    cv.Optional(
+                        CONF_DELAY, default="1s"
+                    ): cv.positive_time_period_milliseconds,
+                    cv.Optional(CONF_BACKOFF_FACTOR, 1.0): cv.float_,
```

Comment:
Similar to above, perhaps setting `1` as the minimum?
```suggestion
                    cv.Optional(CONF_BACKOFF_FACTOR, 1.0): cv.float_range(min=1),
```